homeassistant/components/palazzetti/climate.py

```diff
@@ -0,0 +1,164 @@
+)
+from .coordinator import PalazzettiDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Palazzetti climates based on a config entry."""
+    coordinator: PalazzettiDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    entities: list[PalazzettiClimateEntity] = []
+    await coordinator.async_config_entry_first_refresh()
+    if coordinator.data[AVAILABLE]:
+        entities.append(PalazzettiClimateEntity(coordinator=coordinator))
+        async_add_entities(entities)
+
+
+class PalazzettiClimateEntity(
+    CoordinatorEntity[PalazzettiDataUpdateCoordinator], ClimateEntity
+):
+    """Defines a Palazzetti climate."""
+
+    _attr_has_entity_name = True
+    _attr_hvac_modes = []  # The available modes will be set when we know the current state
+    _attr_fan_modes = FAN_MODES
+    _attr_target_temperature_step = 1.0
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_translation_key = DOMAIN
+
+    def __init__(
+        self,
+        *,
+        coordinator: PalazzettiDataUpdateCoordinator,
+    ) -> None:
+        """Initialize Palazzetti climate."""
+        super().__init__(coordinator=coordinator)
+        self.hub = coordinator.hub
+
+        # TURN_OFF and TURN_ON are not always available. An update coordinator listener will add them when possible.
+        self._attr_supported_features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.FAN_MODE
+        )
+        coordinator.async_add_listener(self._coordinator_update)
+
+        name = self.hub.product.response_json[API_NAME]
+        self._attr_unique_id = coordinator.entry.data[MAC]
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, coordinator.entry.data[MAC])},
+            name=PALAZZETTI,
+            manufacturer=PALAZZETTI,
+            sw_version=self.hub.product.response_json[API_SW_VERSION],
+            hw_version=self.hub.product.response_json[API_HW_VERSION],
+        )
+        self._attr_name = name
+
+    @callback
+    def _coordinator_update(self) -> None:
+        available = (
+            self.coordinator.data["available"]
+            and self.hub
+            and self.hub.hub_online
+            and self.hub.product
+            and self.hub.product_online
+        )
+        self._attr_available = available
+        if available:
+            if self.hub.product.get_data_config_object().flag_has_switch_on_off:
+                self._attr_supported_features |= (
+                    ClimateEntityFeature.TURN_OFF | ClimateEntityFeature.TURN_ON
+                )
+                self._attr_hvac_modes = [HVACMode.OFF, HVACMode.HEAT]
+            else:
+                self._attr_supported_features &= ~(
+                    ClimateEntityFeature.TURN_OFF | ClimateEntityFeature.TURN_ON
+                )
+                self._attr_hvac_modes = [self.hvac_mode]
```

Comment:
I think I'd prefer the properties here as well for readability

Reply:
Addressed in the new revision.